File: libs/shared/kotlin/Activity.kt
package com.sentinel.lifeos.shared.models

data class Activity(
    val id: Int,
    val module_id: Int,
    val type: String,
    val value: Double,
    val metadata: Map<String, Any>? = null,
    val created_at: String
)

data class ActivityCreate(
    val module_id: Int,
    val type: String,
    val value: Double,
    val metadata: Map<String, Any>? = null
)

File: libs/shared/typescript/activity.ts
export interface Activity {
  id: number;
  module_id: number;
  type: string;
  value: number;
  metadata?: Record<string, any>;
  created_at: string;
}

export interface ActivityCreate {
  module_id: number;
  type: string;
  value: number;
  metadata?: Record<string, any>;
}